app/src/main/res/layout/add_habit_dialog_box.xml

```diff
@@ -109,15 +114,185 @@
         android:id="@+id/habit_date_selection"
         android:layout_width="208dp"
         android:layout_height="27dp"
-        android:layout_marginTop="28dp"
+        android:layout_marginTop="32dp"
         android:layout_marginEnd="24dp"
         android:background="@color/white"
         android:clickable="true"
-        android:onClick="onClick"
         android:hint="@string/habit_date"
-        android:textSize="18dp"
+        android:onClick="onClick"
         android:textColor="@color/grey"
+        android:textSize="18dp"
         app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
         app:layout_constraintStart_toStartOf="@+id/habit_reason_edit_text"
         app:layout_constraintTop_toBottomOf="@+id/habit_reason_edit_text" />
+
+    <TextView
+        android:id="@+id/weekly_schedule_header"
+        android:layout_width="132dp"
+        android:layout_height="23dp"
+        android:layout_marginTop="24dp"
+        android:text="Weekly Schedule"
+        app:layout_constraintStart_toStartOf="@+id/habit_date_text_view"
+        app:layout_constraintTop_toBottomOf="@+id/habit_date_text_view" />
+
+    <CheckBox
+        android:id="@+id/monday_button"
+        style="@style/Widget.AppCompat.CompoundButton.CheckBox"
+        android:layout_width="40dp"
+        android:layout_height="40dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="16dp"
+        android:layout_weight="1"
+        android:autoSizeMaxTextSize="36dp"
+        android:autoSizeMinTextSize="6dp"
+        android:autoSizeStepGranularity="2dp"
+        android:autoSizeTextType="uniform"
+        android:background="@drawable/day_button"
+        android:button="@null"
+        android:elevation="5dp"
+        android:enabled="true"
+        android:paddingLeft="8sp"
+        android:paddingRight="8sp"
+        android:text="Mo"
+        android:textColor="@color/black"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/weekly_schedule_header" />
```

Comment:
Few things to notice:
background is set to the day_button drawable, which handles the changes when clicked on
button is null, which gets rid of the check box drawable, leaving the actual background as the checkbox.
The padding varies from button to button to account for letter size, without the padding the buttons have varying sizes of text. This also allows the scaling of text to be consistent in size too.